curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/file-metadata-microservice.english.md

```diff
@@ -23,12 +23,37 @@ Start this project on Repl.it using <a href='https://repl.it/github/freeCodeCamp
 
 ```yml
 tests:
+  - text: I can provide my own project, not the example URL.
+    testString: "getUserInput => {
+      assert(!(new RegExp('.*/file-metadata.freecodecamp.repl.co')).test(getUserInput('url')));
+    }"
   - text: I can submit a form that includes a file upload.
-    testString: ''
+    testString: "async getUserInput => {
+      const site = await fetch(getUserInput('url'));
+      const data = await site.text();
+      assert(/<form.*>/.test(data));
+      assert(/type=['|\"]file['|\"]/.test(data));
+    }"
   - text: The form file input field has the <code>name</code> attribute set to <code>upfile</code>.
-    testString: ''
+    testString: "async getUserInput => {
+      const site = await fetch(getUserInput('url'));
+      const data = await site.text();
+      assert(/name=['|\"]upfile['|\"]/.test(data));
```

Comment:
I like where @darren-stoll was going here, but I think it could be as simple as:
```suggestion
      const doc = new DOMParser().parseFromString(data, 'text/html');
      assert(doc.querySelector('input[name=\"upfile\"]'));
```

Reply:
Okay I've applied this approach to both of the tests that look for HTML elements. 🙂